src/wrapper_array.jl

```diff
@@ -155,24 +155,12 @@ for f in (:getindex, :view, :dotview)
             return Base.$f(A, inds...)
         end
 
-        @propagate_inbounds function Base.$f(a::NamedDimsArray, inds::Vararg{<:Integer})
-            # Easy scalar case, will just return the element
-            return Base.$f(parent(a), inds...)
-        end
-
-        @propagate_inbounds function Base.$f(a::NamedDimsArray, ci::CartesianIndex)
-            # Easy scalar case, will just return the element
-            return Base.$f(parent(a), ci)
-        end
-
-        @propagate_inbounds function Base.$f(a::NamedDimsArray, inds...)
-            # Some nonscalar case, will return an array, probably with names
+        @propagate_inbounds function Base.$f(a::NamedDimsArray, raw_inds...)
+            inds = Base.to_indices(a, raw_inds)
```

Comment:
can I summon @iamed2  for review here.
They understand `to_indices` better than I do, infact when I wrote this code I didn't know it existed.

 think this is a good idea and we should ghave been doing it this way all along, but not 100% sure

Reply:
Not en expert, but I start to think this is the correct official thing to do here. For instance it will mean that https://github.com/mbauman/InvertedIndices.jl works by design, rather than by good luck.

Reply:
I believe that if you're passing through to `getindex` or `view` (`dotview` passes through to either `getindex` or `view` so I'm not sure why you're defining `dotview` here), you can expect `to_indices` to be called for you. I would not call `to_indices` here.

Reply:
`dotview` issue spun off here: https://github.com/invenia/NamedDims.jl/pull/109

Reply:
The point of doing this for `getindex` is that weird index types this package doesn't know about are then normalised, and can be safely handled by `remaining_dimnames_from_indexing`.

(No opinion on `dotview`, it was there a long time.)

Reply:
Here's an example of what I mean about normalisation:
```julia
using NamedDims, InvertedIndices
nda = NamedDimsArray([10 20; 30 40], (:x, :y));
dimnames(nda[Not(CartesianIndex(1,1))]) # (:x,) on tagged version, accidentally

typeof(Not(CartesianIndex(1,1))) # InvertedIndex{CartesianIndex{2}}
supertype(ans) # Any

inds = Base.to_indices(nda, (Not(CartesianIndex(1,1)),))
supertype(typeof(inds[1])) # AbstractArray{CartesianIndex{2},1}
```

Reply:
You then lose the behaviour of any `getindex` that _doesn't_ just depend on `to_indices`. I was worried about that but then I decided those array types should just fix themselves.

So this is probably fine then. Normalizing for `remaining_dimnames_from_indexing` makes sense.

Reply:
Great, then perhaps this is done.